src/AutomationTests/AutomationIntegrationTests.cs

```diff
@@ -143,53 +145,118 @@ public void ScanAsync_WindowsFormsSampler_TaskIsCancelled_ThrowsCancellationExce
+        private WindowScanOutput ScanIntegrationCore(bool sync, string testAppPath, int expectedErrorCount, int expectedWindowCount = 1, int? processId = null)
         {
-            LaunchTestApp(testAppPath);
-            var builder = Config.Builder.ForProcessId(TestProcess.Id)
+            if (processId == null)
+            {
+                processId = LaunchTestApp(testAppPath);
+            }
+            var builder = Config.Builder.ForProcessId((int)processId)
                 .WithOutputDirectory(OutputDir)
                 .WithOutputFileFormat(OutputFileFormat.A11yTest);
 
             var config = builder.Build();
 
             var scanner = ScannerFactory.CreateScanner(config);
 
-            var output = ScanSyncWithProvisionForBuildAgents(scanner);
+            IReadOnlyCollection<WindowScanOutput> output;
+            if (sync)
+            {
+                output = ScanSyncWithProvisionForBuildAgents(scanner);
+            }
+            else
+            {
+                output = ScanAsyncWithProvisionForBuildAgents(scanner);
+            }
 
             return ValidateOutput(output, expectedErrorCount, expectedWindowCount);
         }
 
-        private WindowScanOutput AsyncScanIntegrationCore(string testAppPath, int expectedErrorCount, int expectedWindowCount = 1)
+        private IEnumerable<Task<ScanOutput>> GetAsyncScanTasks(string testAppPath, IEnumerable<CancellationToken> cancellationTokens)
         {
-            LaunchTestApp(testAppPath);
-            var builder = Config.Builder.ForProcessId(TestProcess.Id)
-                .WithOutputDirectory(OutputDir)
-                .WithOutputFileFormat(OutputFileFormat.A11yTest);
+            var taskFuncs = new List<Func<Task<ScanOutput>>>();
 
-            var config = builder.Build();
+            // Prepare scan tasks
+            foreach (var token in cancellationTokens)
+            {
+                var processId = LaunchTestApp(testAppPath);
+                var builder = Config.Builder.ForProcessId(processId)
+                    .WithOutputDirectory(OutputDir)
+                    .WithOutputFileFormat(OutputFileFormat.A11yTest);
 
-            var scanner = ScannerFactory.CreateScanner(config);
+                var config = builder.Build();
 
-            var output = ScanAsyncWithProvisionForBuildAgents(scanner);
+                var scanner = ScannerFactory.CreateScanner(config);
 
-            return ValidateOutput(output, expectedErrorCount, expectedWindowCount);
+                taskFuncs.Add(() => scanner.ScanAsync(null, token));
+            }
+
+            // Kick scan tasks off
+            var tasks = new List<Task<ScanOutput>>();
+            foreach (var func in taskFuncs)
+            {
+                tasks.Add(func());
+                Thread.Sleep(300); // TODO investigate why this is necessary (without this UIAutomation.GetRootElement() throws)
```

Comment:
@codeofdusk @DaveTryon do either of you recognize the following error for UIAutomation? It's getting thrown if you start `ScanAsync` tasks too close together. I've added this thread.sleep for now while I wrote the tests but I strongly suspect that this is what's causing the error in CI. 

> Error HRESULT E_FAIL has been returned from a call to a COM component.

Reply:
Looks like UIAutomation has some odd multithreaded behavior, but adding a lock around those calls fixed it for me locally (see [c148be8](https://github.com/microsoft/axe-windows/pull/764/commits/c148be8ffa8186be41c110f43574fb9fc04dd13c))